src/native/corehost/hostmisc/pal.h

```diff
@@ -293,15 +293,34 @@ namespace pal
     bool getenv(const char_t* name, string_t* recv);
     bool get_default_servicing_directory(string_t* recv);
 
-    // Returns the globally registered install location (if any)
+    enum class known_architecture
```

Comment:
I'd recommend matching the same order as the `Architecture` enum.

https://github.com/dotnet/runtime/blob/7066ca90620635c36b94391ad7a09a85a29739ac/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Architecture.cs#L6-L16

Reply:
I went for alphabetical order for ease of iterating over and printing out.
As in you'd want them to have the same values? (There's no wasm in this case.)

Reply:
It is fine if wasm isn't used at present. I find keeping these enum patterns identical is helpful as it reminds people where all to update when one thing is updated. I think WASM in this case is an exception rather than the rule, but that is what the `default` case in a switch is for :)

Reply:
Spoke offline. This makes more sense.